stm32/libcanard/bxcan/src/bxcan.c

```diff
@@ -0,0 +1,854 @@
+/// This software is distributed under the terms of the MIT License.
+/// Copyright (c) 2016-2020 UAVCAN Development Team.
+/// Authors: Pavel Kirienko <pavel.kirienko@zubax.com>, Tom De Rybel <tom.derybel@robocow.be>
+
+#include "bxcan.h"
+#include "bxcan_registers.h"
+#include <assert.h>
+#include <string.h>
+
+// bxCAN driver user configuration.
+#define BXCAN_MAX_IFACE_INDEX 0U                            // Single CAN interface, thus maximum iface index = 0.
+#define BXCAN_BUSYWAIT_DELAY_SYSTEM_CORE_CLOCK 80000000UL   // Processor core clock is 80.000.000 Hz.
```

Comment:
This is out of place here. The user should not be required to modify the driver to use it. Instead, the parameters are to be passed using preprocessor flags, like `-DBXCAN_BUSYWAIT_DELAY_SYSTEM_CORE_CLOCK=80000000UL`, or by pre-including a header like `-imyheader.h`.

Reply:
Thanks for explaining how to do this. I was going to look into it later.